nav2_controller/src/controller_server.cpp

```diff
@@ -604,14 +615,7 @@ void ControllerServer::setPlannerPath(const nav_msgs::msg::Path & path)
     throw nav2_core::InvalidPath("Path is empty.");
   }
   controllers_[current_controller_]->setPlan(path);
```

Comment:
Do we need this `setPlan` if its now being given in the compute velocity command context? We do sometimes do other things in the setPlan like reset state, but maybe then it could be refactored or done in another way?

Reply:
> MPPI and DWB needs to know the last pose of the plan (goal)

If the computation API has this, it can be set there now I think. 

> Reset state when new plan receives

Yes I agree. 

---

Maybe what makes sense is to change the API to be more of a notification about a new path (and the raw, full global path path can be given) and we change the function name / description. But, we don't use that path for anything that the raw, global framed, full path isn't necessary for. We use the path given in the compute velocity command API instead which is pre-transformed and pruned.